Orange/widgets/unsupervised/owsom.py

```diff
@@ -828,13 +828,19 @@ def update_output(self):
 
     def set_color_bins(self):
         if self.attr_color is None:
-            self.thresholds = self.colors = None
+            self.thresholds = self.bin_labels = self.colors = None
         elif self.attr_color.is_discrete:
-            self.thresholds = None
+            self.thresholds = self.bin_labels = None
             self.colors = [QColor(*color) for color in self.attr_color.colors]
         else:
             col = self.data.get_column_view(self.attr_color)[0].astype(float)
-            self.thresholds = decimal_binnings(col, min_bins=4)[0][1:-1]
+            binning = decimal_binnings(col, min_bins=4)[-1]
```

Comment:
I think you forget to check whether self.attr_color is a Time Variable. This doesn't use time_binnings.

Reply:
Oops. I just ported it to new signature of `decimal_binning`. It works now.

I'm not sure I like the legend. It uses short labels.

<img width="291" alt="Screenshot 2019-10-21 at 16 32 15" src="https://user-images.githubusercontent.com/2387315/67214627-5abd5880-f420-11e9-91e8-4d28f272a356.png">

Perhaps the lower bound for each interval should be in the long form. That is, the last two should be `17 Apr - Jul` and `>= 17 Jul`. The second would be in short form, except of course when the longer is needed, like in `16 Oct - 17 Jan`.

Reply:
Like this.

<img width="275" alt="Screenshot 2019-10-21 at 19 27 15" src="https://user-images.githubusercontent.com/2387315/67228039-c7dce800-f438-11e9-9638-526678d8e5f2.png">